contribute/design/proposal-volume-snapshots-clone.md

```diff
@@ -46,21 +52,32 @@ The usecases related to protecting data against application failures and recover
 - Delete Clone Volume
 - Revert Volume State to a previous Snapshot
 
-Along wih the OpenEBS Volume Dynamic Provisioner, OpenEBS Snapshot Provisioner will be launched during the openebs install. The Snapshot Provisioner is an extension/plugin of https://github.com/kubernetes-incubator/external-storage/tree/master/snapshot
+When installing openebs, OpenEBS Snapshot Provisioner will be launched. The OpenEBS Snapshot Provisioner is an extension/plugin of https://github.com/kubernetes-incubator/external-storage/tree/master/snapshot
 
 ### Create Snapshot
 * Admin uses the kubectl volume snapshot yaml to create a snapshot on a given PV/PVC. 
-* Kubernets will forward the snapshot create request to the OpenEBS Snapshot provisioner, while will do the following:
+* Kubernetes will forward the snapshot create request to the OpenEBS Snapshot provisioner, which will do the following:
   - Identify the cStorService or the JivaService associated with the PV
   - If Jiva, call the REST API of Jiva Controller to create a new snapshot
-  - If cStor, call the REST API of cStorVolumeMgmt(side-car?) to create a new snapshot
+  - If cStor: 
+    * cstor-volume-mgmt side-car will be provisioned for cStorController. This will listen for cStorVolumeSnapshot CR events
+    * cStorVolumeSnasphot will contain details like:
+      - Volume Identifier
+      - Snapshot Name
+      - Status Fields for Controller and Replicas will be initialized with "init". Only the Replicas that are online will be included.
+      (TODO: Vitta/Jeffry - If one of the Replica was offline during the snapshot creation, will it be created during rebuild process?)
+    * cstor-volume-mgmt Add Event handler will pause the IOs and update the contoller state to pause. The controller will wait for a max of 2 seconds(?) for the snapshot to be taken on all the replicas. After every 1000ms, this side-car will check to resume IOs, if all the replica status are updated. 
+    * cstor-pool-mgmt side-car will wait for the controller to set its status to pause and they will create Snapshot on the corresponding volume. After taking the snapshot, the status field will be updated. 
+    * OpenEBS Snapshot Provisioner will check on the status of the cStorVolumeSnapshot CR and marks the success/failure.
 * OpenEBS Snapshot provisioner will return the status back to Kubernetes
 * Kubernetes will store the snapshot against the PV. The list of created snapshots can be obtained using kubectl get snapshots.
 
 ### Delete Snapshot
 * Admin uses the kubectl volume snapshot yaml to create a snapshot on a given PV/PVC. 
 * Kubernetes will delete snapshot from its configuration. Also pass the request to the OpenEBS Snapshot provisioner for cleanup. 
-* OpenEBS Snapshot provisioner will check if the volume associated is of type cStor and will forward the delete snapshot request to cStorVolumeMgmt(side-car?)
+* If the volume type is cStor:
```

Comment:
Deletion can be allowed even if few replicas are down. But, it need to be retied on them once those replicas comes back.